resource/resource.go

```diff
@@ -168,46 +151,145 @@ func (r *Resource) Out(inputDirectory string, source oc.Source, params oc.Params
 	if err != nil {
 		return nil, nil, err
 	}
-
 	p, err := parseOutParams(params)
 	if err != nil {
 		return nil, nil, err
 	}
-
 	err = r.configureClient(s)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	rootDir := filepath.Join(inputDirectory, p.Path)
 
-	err = filepath.Walk(rootDir,
-		func(path string, info os.FileInfo, err error) error {
-			if info.IsDir() {
-				return nil
+	for _, secretMap := range p.SecretMaps {
+
+		sourceDir := filepath.Join(rootDir, secretMap.Source)
+		files, err := listFilesUnder(sourceDir)
```

Comment:
this approach effectively semi-supports recursion, in that you could specify a partial path, and this would loop over all the children. However, it doesn't support all the validation and making sense of how renaming would work for such a feature. Therefore, it would probably be best to not support partial paths, and then to also move this filetree walk out of the loop, and just know what files you have at your disposal before entering the loop in the first place.